egs/wsj/s5/steps/nnet3/components.py

```diff
@@ -186,13 +196,34 @@ def AddFinalLayer(config_lines, input, output_dim,
     if include_log_softmax:
         if use_presoftmax_prior_scale :
             components.append('component name={0}-fixed-scale type=FixedScaleComponent scales={1}'.format(final_node_prefix, prior_scale_file))
-            component_nodes.append('component-node name=Final-fixed-scale component={0}-fixed-scale input={1}'.format(finale_node_prefix,
+            component_nodes.append('component-node name={0}-fixed-scale component={0}-fixed-scale input={1}'.format(final_node_prefix,
                 prev_layer_output['descriptor']))
             prev_layer_output['descriptor'] = "{0}-fixed-scale".format(final_node_prefix)
         prev_layer_output = AddSoftmaxLayer(config_lines, final_node_prefix, prev_layer_output)
     # we use the same name_affix as a prefix in for affine/scale nodes but as a
     # suffix for output node
-    AddOutputLayer(config_lines, prev_layer_output, label_delay, suffix = name_affix)
+    AddOutputLayer(config_lines, prev_layer_output, label_delay, suffix = name_affix, objective_type = objective_type)
+
+def AddFinalSigmoidLayer(config_lines, input, output_dim,
```

Comment:
You should add a comment on what a FinalSigmoidLayer is and where it is intended to be used. I think the default objective_type for this function should be quadratic, as it will used primarily in that context.